### PR TITLE
Features/collapse writer activities

### PR DESCRIPTION
**Pull request by Wu-Jing**
Remove redundant database reader activity.
Collapse database writer activity.
Utilize spark JDBC dialect for sql statement column name.
Move shared sink logic to base class.

### CODE REVIEW

File: pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseWriterActivity.scala [hunk 1]
```diff
@@ -3,33 +3,50 @@ package yumi.pipeline.activities
...
     .getAs[Map[String, Any]]("properties")
     .map {
       case (key, value) =>
         (key -> value.toString)
     }
 
+  //looking for some better way to write this
+  val databaseType = url.split(":")(1)
+
+  val jdbcDriverClass = databaseType match
```
Comment (coderdiaries) [inline pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseWriterActivity.scala]: The driver needs to be additional parameter now.

File: pipeline/src/test/scala/test/yumi/pipeline/activities/SqlDatabaseWriterActivitySpec.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    val dataFrame = mock[DataFrame]
+
+    when(sparkSession.table(table)).thenReturn(dataFrame)
+
+    // Act
+    val activity = new SqlDatabaseWriterActivity(parameters)
+    val resultSessionData = activity.invoke(sessionData)
+
+    // Assert
+
```
Comment (coderdiaries) [inline pipeline/src/test/scala/test/yumi/pipeline/activities/SqlDatabaseWriterActivitySpec.scala]: need to verify dataFrameWriter invocation here.

File: pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseReaderActivity.scala [hunk 3]
```diff
@@ -4,10 +4,11 @@ import java.util.Properties
 
 import yumi.pipeline.{Activity, Parameters, SessionData, YumiContext}
 
-class PostgreSqlReaderActivity (val parameters: Parameters) extends Activity{
+class SqlDatabaseReaderActivity (val parameters: Parameters) extends Activity {
```
Comment (coderdiaries) [inline pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseReaderActivity.scala]: Lets rename this to `DatabaseReaderActivity`

File: pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseWriterActivity.scala [hunk 4]
```diff
@@ -3,33 +3,50 @@ package yumi.pipeline.activities
 import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions
 import yumi.pipeline.{Activity, Parameters, SessionData, YumiContext}
 
-class PostgreSqlWriterActivity(val parameters: Parameters) extends Activity  {
-
+class SqlDatabaseWriterActivity (val parameters: Parameters) extends Activity {
```
Comment (coderdiaries) [inline pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseWriterActivity.scala]: Lets rename this to `DatabaseWriterActivity`

File: pipeline/src/main/scala/yumi/pipeline/sinks/DbSink.scala [hunk 5]
```diff
@@ -1,9 +1,14 @@
...
+
 import org.apache.spark.sql.execution.datasources.jdbc.{JDBCOptions, JdbcUtils}
-import org.apache.spark.sql.jdbc.JdbcDialects
+import org.apache.spark.sql.jdbc.{JdbcDialect, JdbcDialects}
+import org.apache.spark.sql.types._
 import org.apache.spark.sql.{DataFrame, Row}
 
+import scala.util.control.NonFatal
+
 abstract class DbSink extends GenericSink {
```
Comment (coderdiaries) [inline pipeline/src/main/scala/yumi/pipeline/sinks/DbSink.scala]: Lets rename this to `DatabaseSync`

File: pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseWriterActivity.scala (hunk 1 above)
Comment (Wu-Jing) [inline pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseWriterActivity.scala]: Okay. I was thinking the database type information is already in the url, so we don't want the user to specify database type again for the driver.

File: pipeline/src/test/scala/test/yumi/pipeline/activities/SqlDatabaseWriterActivitySpec.scala (hunk 2 above)
Comment (Wu-Jing) [inline pipeline/src/test/scala/test/yumi/pipeline/activities/SqlDatabaseWriterActivitySpec.scala]: will update.

File: pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseReaderActivity.scala (hunk 3 above)
Comment (Wu-Jing) [inline pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseReaderActivity.scala]: Will update.

File: pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseWriterActivity.scala (hunk 4 above)
Comment (Wu-Jing) [inline pipeline/src/main/scala/yumi/pipeline/activities/SqlDatabaseWriterActivity.scala]: Will update.

File: pipeline/src/main/scala/yumi/pipeline/sinks/DbSink.scala (hunk 5 above)
Comment (Wu-Jing) [inline pipeline/src/main/scala/yumi/pipeline/sinks/DbSink.scala]: Will update.

**Comment by Wu-Jing** [comment]
@coderdiaries  Updated based on comments.

**Pull request merged**